packages/blade/src/components/BottomSheet/_decisions/decisions.md

```diff
@@ -0,0 +1,245 @@
+
+## Anatomy
+
+<img src="./bottomsheet-anatomy.png" alt="BottomSheet Anatomy" width="100%" />
+
+
+Components:
+
+- BottomSheet
+  - BottomSheetHeader
+  - BottomSheetBody
+  - BottomSheetFooter
+
+## API
+
+Sample usage:
+
+```jsx
+import { BottomSheet } from '@razorpay/blade';
+
+<BottomSheet isOpen={boolean} snapPoints={[]}>
+  <BottomSheetHeader
+    title="Select Account"
+    leading={<BladeIcon />}
+  />
+  <BottomSheetBody>
+    <Text>Body Content<Text>
+  </BottomSheetBody>
+  <BottomSheetFooter
+    title="Footer Title"
+    leading={BladeIcon}
+    trailing={
+      <>
+        BladeIcon | IconButton | Button
+      </>
+    }
+  />
+</BottomSheet>;
+```
+
+**Design Constraints**
+
+- BottomSheetHeader:
+  - title
+  - leading:
+    - BladeIcon
+
+- BottomSheetFooter:
+  - title
+  - leading:
+    - BladeIcon
+  - trailing
+    - any jsx element
+
+- The BottomSheetFooter will behave similar to ActionListFooter.
+- The Header, Footer can be individually omitted
+
+### `BottomSheet`
+
+We'll expose a `BottomSheet` component with the following API:
+
+| Prop            | Type        | Default                 | Description                                                                                     | Required |
+| --------------- | ----------- | ----------------------- | ----------------------------------------------------------------------------------------------- | -------- |
+| isOpen            | `boolean`   | `false`                 | Toggles bottom sheet state                                                              |          |
+| snapPoints      | `[number, number, number]`  | `[0.35, 0.5, 0.85]` | Snappoints in which the bottom sheeet will rest on, this accepts a number between 0 & 1 which maps to the total view height of the screen. 0.5 means 50% of screen height.                                              |          |
+| onDismiss       | `Callback`  | `undefined`             | called when the bottom sheet is closed, either by user state, hitting `esc` or tapping backdrop |          |
+| initialFocusRef | `React.Ref` | `undefined`             | ref element you want to get keyboard focus when opening the sheet                               |          |
+
+### `BottomSheetBody`
+
+| Prop       | Type              | Default     | Description                 | Required |
+| ---------- | ----------------- | ----------- | --------------------------- | -------- |
+| `children` | `React.ReactNode` | `undefined` | Contents of the BottomSheet | ✅       |
+
+### `BottomSheetHeader` API
+
+| Prop     | Type            | Default     | Description                          | Required |
+| -------- | --------------- | ----------- | ------------------------------------ | -------- |
+| `title`  | `string`        | `undefined` | Title of the Header                    | ✅       |
+| `leading` | `IconComponent` | `undefined` | leading icon placed before title text |          |
+
+### `BottomSheetFooter` API
```

Comment:
just initial thoughts:

- seems unnecessarily constraint-based to me? 
- the fact that I need to use a trailing prop to render 2 full-width action buttons has a steep learning curve and is confusing because it doesn't match the layout visually. I wouldn't have assumed that to be in trailing
- next, the fact that leading is always an icon component is not very justifiable. it feels like its meant to be used with the title

what if we make footer flexible to any jsx element? this might even come up when we are generalising Header and Footer components across our components. Lets brainstorm this with design team

Reply:
imagine use cases for checkout

where they'll have one action button saying "Pay Now" towards the right/trailing and towards leading they might show things like "Pay Using XX32" card number(like we see in zomato)

Reply:
> what if we make footer flexible to any jsx element?

Flexible hi to hain? the trailing can take any JSX element, the API is 1:1 with ActionListFooter. 

Are you saying make the whole footer accept any jsx?